course3/homework-string/my_string.cc

```diff
@@ -0,0 +1,105 @@
+#include <iostream>
+#include <cstring>
+#include "my_string.h"
+
+MyString::MyString(){
+	buffer = std::unique_ptr<char[]>(new char[1]);
+	buffer[0] = '\0';
+}
+
+MyString::MyString(const char* theString){
+	buffer = std::unique_ptr<char[]>(new char[std::strlen(theString) + 1]);
+	unsigned int i;
+	for (i = 0; i < std::strlen(theString); i++){
+			buffer[i] = theString[i];
+	}
+	buffer[i] = '\0';
+}
+
+MyString::MyString(const MyString& theString){
+	buffer = std::unique_ptr<char[]>(new char[theString.size() + 1]);
+	unsigned int i;
+	for (i = 0; i < std::strlen(theString.buffer.get()); i++){
+		buffer[i] = theString[i];
+	}
+	buffer[i] = '\0';
+}
+   
+const bool MyString::empty() const {
+	if (buffer){
```

Comment:
can buffer ever be null in your string? It looks like whichever path you take, it should never be null, so the check is not needed